tests/admin/test-class-admin-editor-specific-replace-vars.php

```diff
@@ -8,247 +8,250 @@
+
+	/**
+	 * Tests that determine_for_term can detect a post_format.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_term
+	 */
+	public function test_determine_for_term_with_a_post_format() {
+		$this->assertEquals( 'post_format', $this->class_instance->determine_for_term( 'post_format' ) );
+	}
+
+	/**
+	 * Tests that determine_for_post defaults to post when no actual post variable is passed along.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_without_a_wp_post_instance() {
+		$this->assertEquals( 'post', $this->class_instance->determine_for_post( null ) );
+	}
+
+	/**
+	 * Tests that a homepage is succesfully determined.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::is_homepage
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_homepage() {
+		// Backup the current values for these options.
+		$show_on_front = get_option( 'show_on_front' );
+		$page_on_front = get_option( 'page_on_front' );
+
+		$post = $this->create_and_get_with_post_type( 'page' );
+
+		// Overwrite the options used in the is_homepage function.
+		update_option( 'show_on_front', 'page' );
+		update_option( 'page_on_front', $post->ID );
+
+		$this->assertEquals( 'page', $this->class_instance->determine_for_post( $post ) );
+
+		// Revert the options their original values.
+		update_option( 'show_on_front', $show_on_front );
+		update_option( 'page_on_front', $page_on_front );
+	}
+
+	/**
+	 * Tests that determine_for_post can detect a page.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_page() {
+		$post = $this->create_and_get_with_post_type( 'page' );
+
+		$this->assertEquals( 'page', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Tests that determine_for_post can detect a post.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_post() {
+		$post = $this->create_and_get_with_post_type( 'post' );
+
+		$this->assertEquals( 'post', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Tests that determine_for_post can detect a custom post type.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post
+	 */
+	public function test_determine_for_post_with_a_custom_post_type() {
+		$post = $this->create_and_get_with_post_type( 'some_plugin_post' );
+
+		$this->assertEquals( 'custom_post_type', $this->class_instance->determine_for_post( $post ) );
+	}
+
+	/**
+	 * Tests that determine_for_post_type works for a post.
+	 *
+	 * @covers WPSEO_Admin_Editor_Specific_Replace_Vars::determine_for_post_type
+	 */
+	public function test_determine_for_post_type_with_a_post() {
+		$this->assertEquals( 'post', $this->class_instance->determine_for_post_type( 'post' ) );
```

Comment:
Please consider using a dataProvider for testing the `determine_for_post_type` method